staging/src/k8s.io/apimachinery/pkg/runtime/scheme.go

```diff
@@ -501,6 +567,31 @@ func (s *Scheme) convertToVersion(copy bool, in Object, target GroupVersioner) (
 	return out, nil
 }
 
+// convertUnstructuredToVersioned attempts to convert an unstructured object to a typed
+// object if possible. It will return false if conversion is not possible, or the versioned
+// Go form of the object. Note that this conversion will lose fields.
+func (s *Scheme) convertUnstructuredToVersioned(in Unstructured) (Object, error) {
```

Comment:
better name... this doesn't convert between two different gvk's, it converts unstructured to typed